src/components/select/select.js

```diff
@@ -74,7 +74,8 @@ angular.module('material.components.select', [
  * @param {expression=} md-selected-text Expression to be evaluated that will return a string
  * to be displayed as a placeholder in the select input box when it is closed.
  * @param {string=} placeholder Placeholder hint text.
- * @param md-no-asterisk {boolean=} When set to true, an asterisk will not be appended to the floating label.
+ * @param md-no-asterisk {boolean=} When set to true, an asterisk will not be appended to the
```

Comment:
I have always pushed for the opt-in approach; however, this has been around since before 1.0 (before I joined the team I think) so I don't think we should change the API at this point.

All I did here was add a note about it only being evaluated once; not being watched.